lib/block_view_validator.go

```diff
@@ -1772,25 +1862,20 @@ func (bav *UtxoView) GetTopActiveValidatorsByStake(limit int) ([]*ValidatorEntry
 }
 
 func (bav *UtxoView) GetGlobalStakeAmountNanos() (*uint256.Int, error) {
-	var globalStakeAmountNanos *uint256.Int
-	var err error
 	// Read the GlobalStakeAmountNanos from the UtxoView.
 	if bav.GlobalStakeAmountNanos != nil {
-		globalStakeAmountNanos = bav.GlobalStakeAmountNanos.Clone()
+		return bav.GlobalStakeAmountNanos, nil
 	}
 	// If not set, read the GlobalStakeAmountNanos from the db.
-	// TODO: Confirm if the GlobalStakeAmountNanos.IsZero() that we should look in the db.
```

Comment:
Sanity check caught that we *don't* want to go to the db if the `UtxoView.GlobalStakeAmountNanos.IsZero()`. To illustrate, imagine a case where the existing global stake is 1 $DESO, but the one and only validator unstakes resulting in a global stake in the UtxoView of 0 $DESO. If we fetch the current global stake, we will see it's zero in the UtxoView so go to the database where it's still 1 $DESO, because we haven't flushed the UtxoView to the db yet. For that reason, we only want to go to the db if the global stake is `nil` in the UtxoView.

Reply:
Wow did I miss this? Good catch guys.